### PR TITLE
Finish catalog

### PR DESCRIPTION
**Pull request by eugene-chr**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{
+  "name": "marvel-catalog",
+  "version": "1.0.0",
+  "private": true,
+  "dependencies": {
+    "@material-ui/core": "^4.9.12",
+    "@material-ui/icons": "^4.9.1",
+    "@material-ui/lab": "^4.0.0-alpha.52",
+    "eslint": "^6.8.0",
```
Comment (plzen) [inline package.json:9]: Move dev dependencies to `devDependencies` section

File: src/components/App/component.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  <Router>
+    <Header />
+    <Container maxWidth="md" component="main" className="wrapper">
+      <Switch>
+        <Route exact path={MAIN_PAGE_ROUTE} component={MainPage} />
+
+        <Route exact path={CHARACTERS_ROUTE} component={ListPage} />
+        <Route exact path={COMICS_ROUTE} component={ListPage} />
+
+        <Route exact path="/characters/:id" component={CharacterPage} />
```
Comment (plzen) [inline src/components/App/component.js:31]: Move routes to constants

File: src/components/App/component.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        <Route exact path="/comics/:id" component={ComicPage} />
+
+        <Route path={ABOUT_US_ROUTE} component={AboutUsPage} />
+        <Route path={CONTACTS_ROUTE} component={ContactsPage} />
+
+        <Route path="*" component={NotFoundPage} />
```
Comment (plzen) [inline src/components/App/component.js:37]: You can skip `path` prop, Switch takes last route as default if it isn't matched

File: src/components/shared/ActiveRouterLink/component.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Link, useRouteMatch } from 'react-router-dom';
+import Button from '@material-ui/core/Button';
+
+import { MAIN_PAGE_ROUTE } from 'constants/routes';
+
+const ActiveRouterLink = ({ to, children }) => {
+  const match = useRouteMatch({
```
Comment (plzen) [inline src/components/shared/ActiveRouterLink/component.js:9]: If you follow splitting components to stateful and stateless components, you should use hooks in container only

File: src/components/shared/Pagination/component.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import { Pagination, PaginationItem } from '@material-ui/lab';
+import Box from '@material-ui/core/Box';
+import { Link } from 'react-router-dom';
+
+const PaginationComponent = ({ total, page, path }) => (
+  <Box display="flex" justifyContent="center" alignItems="center" py={4}>
+    <Pagination
+      count={total}
+      size="large"
+      page={+page === 0 ? 1 : +page}
```
Comment (plzen) [inline src/components/shared/Pagination/component.js:12]: This logic is better to use in container

File: src/components/shared/Pagination/component.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      renderItem={(item) => (
+        <PaginationItem
+          component={Link}
+          to={`${path}${page === 1 ? '' : `?page=${item.page}`}`}
```
Comment (plzen) [inline src/components/shared/Pagination/component.js:16]: I think it's ok if you assign page param for first page also

File: src/components/shared/TextBadge/component.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import Badge from '@material-ui/core/Badge';
+import withStyles from '@material-ui/core/styles/withStyles';
+
+const TextBadge = withStyles(() => ({
```
Comment (plzen) [inline src/components/shared/TextBadge/component.js:4]: HOCs are better to use in containers

File: package.json (hunk 1 above)
Comment (eugene-chr) [reply package.json:9]: done

File: src/components/App/component.js (hunk 2 above)
Comment (eugene-chr) [reply src/components/App/component.js:31]: done

File: src/components/App/component.js (hunk 3 above)
Comment (eugene-chr) [reply src/components/App/component.js:37]: done

File: src/components/shared/ActiveRouterLink/component.js (hunk 4 above)
Comment (eugene-chr) [reply src/components/shared/ActiveRouterLink/component.js:9]: done

File: src/components/shared/Pagination/component.js (hunk 5 above)
Comment (eugene-chr) [reply src/components/shared/Pagination/component.js:12]: done

File: src/components/shared/Pagination/component.js (hunk 6 above)
Comment (eugene-chr) [reply src/components/shared/Pagination/component.js:16]: done

**Pull request merged**